pkg/volume/iscsi/iscsi_util.go

```diff
@@ -623,13 +654,22 @@ func (util *ISCSIUtil) DetachDisk(c iscsiDiskUnmounter, mntPath string) error {
 
 	// If device is no longer used, see if need to logout the target
 	if isSessionBusy(c.iscsiDisk.plugin.host, portals[0], iqn) {
+		err = util.removeISCSI(mntPath)
+		if err != nil {
+			return fmt.Errorf("failed to remove metadata file: %s", err)
+		}
 		return nil
 	}
 
 	err = util.detachISCSIDisk(c.exec, portals, iqn, iface, volName, initiatorName, found)
 	if err != nil {
 		return fmt.Errorf("failed to finish detachISCSIDisk, err: %v", err)
 	}
+	err = util.removeISCSI(mntPath)
+	if err != nil {
+		return fmt.Errorf("failed to remove metadata file: %s", err)
```

Comment:
nit: `%s` while (it seems like) existing code uses `%v`

Reply:
It's really mixed in the code, but I added `%v` to couple of places.